libraries/AC_Autorotation/AC_Autorotation.cpp

```diff
@@ -251,54 +321,121 @@ float AC_Autorotation::get_rpm(bool update_counter)
-// @Field: hserr: head speed error; difference between current and desired head speed
-// @Field: ColOut: Collective Out
-// @Field: FFCol: FF-term for headspeed controller response
-// @Field: CRPM: current headspeed RPM
-// @Field: SpdF: current forward speed
-// @Field: CmdV: desired forward speed
-// @Field: p: p-term of velocity response
-// @Field: ff: ff-term of velocity response
-// @Field: AccO: forward acceleration output
-// @Field: AccT: forward acceleration target
-// @Field: PitT: pitch target
+    // @LoggerMessage: AROT
+    // @Vehicles: Copter
+    // @Description: Helicopter AutoRotation information
+    // @Field: TimeUS: Time since system startup
+    // @Field: P: P-term for headspeed controller response
+    // @Field: hs_e: head speed error; difference between current and desired head speed
+    // @Field: C_Out: Collective Out
+    // @Field: FFCol: FF-term for headspeed controller response
+    // @Field: SpdF: current forward speed
+    // @Field: DH: desired forward speed
+    // @Field: p: p-term of velocity response
+    // @Field: ff: ff-term of velocity response
+    // @Field: AccZ: average z acceleration
+    // @Field: DesV: Desired Sink Rate
+    // @Field: Rfnd: rangefinder altitude
+    // @Field: Hest: estimated altitude
 
     //Write to data flash log
     AP::logger().WriteStreaming("AROT",
-                       "TimeUS,P,hserr,ColOut,FFCol,CRPM,SpdF,CmdV,p,ff,AccO,AccT,PitT",
-                         "Qffffffffffff",
-                        AP_HAL::micros64(),
-                        (double)_p_term_hs,
-                        (double)_head_speed_error,
-                        (double)_collective_out,
-                        (double)_ff_term_hs,
-                        (double)_current_rpm,
-                        (double)_speed_forward,
-                        (double)_cmd_vel,
-                        (double)_vel_p,
-                        (double)_vel_ff,
-                        (double)_accel_out,
-                        (double)_accel_target,
-                        (double)_pitch_target);
+                                "TimeUS,P,hs_e,C_Out,FFCol,SpdF,DH,p,ff,AccZ,DesV,Rfnd,Hest",
+                                "Qffffffffffff",
+                                AP_HAL::micros64(),
+                                (double)_p_term_hs,
+                                (double)_head_speed_error,
+                                (double)_collective_out,
+                                (double)_ff_term_hs,
+                                (double)(_speed_forward*0.01f),
+                                (double)(_cmd_vel*0.01f),
+                                (double)_vel_p,
+                                (double)_vel_ff,
+                                (double)_avg_acc_z,
+                                (double)_desired_sink_rate,
+                                (double)(_radar_alt*0.01f),
+                                (double)(_est_alt*0.01f)) ;
```

Comment:
Existing issue, but you should not need to cast to double here.

Reply:
done